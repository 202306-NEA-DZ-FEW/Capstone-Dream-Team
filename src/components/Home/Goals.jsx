import React from "react";
export default function Goals() {
    const images = [
        {
            url: "images/home/Goals/noPoverty.png",
            title: "Eradicate Poverty",
            text: "We aim to reduce poverty by ensuring that individuals and families in need have access to nutritious meals, regardless of their financial circumstances. Through your generous contributions, we strive to improve the lives of those facing economic hardship",
        },
        {
            url: "images/home/Goals/zeroHunger.png",
            title: "Zero Hunger",
            text: "Our mission is to eliminate hunger by providing regular access to wholesome and nourishing meals for those who are food-insecure. We encourage community participation and volunteering to create a lasting impact on food accessibility.",
        },
        {
            url: "images/home/Goals/health.png",
            title: "Promote Good Health",
            text: "At Buy Me a Meal, we are on a mission to provide nutrition to prevent bad health and combat diseases caused by malnutrition",
        },
    ];

    return (
        <div className='w-full h-full flex-col mt-6 p-6 justify-center  text-center'>
            <h1 className='font-bold mr-6 text-2xl '>Our Goals </h1>
            <div className='flex flex-col items-center md:flex-row w-full h-full md:justify-center md:gap-10 '>
                {images.map((image, index) => (
                    <div
                        key={index}
                        class=' w-1/2 h-full py-4 px-8 bg-white shadow-lg rounded-lg my-20'
                    >
                        <div class='flex justify-center md:justify-end -mt-16'>
                            <img
                                class='w-20 h-20 object-cover rounded-full border-2 border-indigo-500'
                                src={image.url}
                                alt='step'
                            />
                        </div>
                        <div>
                            <h2 class='text-gray-800 text-3xl font-semibold'>
                                {image.title}
                            </h2>
                            <p class='mt-2 text-gray-600 h-full'>
                                {image.text}
                            </p>
                        </div>
                    </div>
                ))}
            </div>
        </div>
    );
}
