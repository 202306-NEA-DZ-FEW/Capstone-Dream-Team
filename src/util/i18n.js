import i18n from "i18next";
import arTranslations from "public/locales/ar/common.json";
import enTranslations from "public/locales/en/common.json";
import frTranslations from "public/locales/fr/common.json";
import { initReactI18next } from "react-i18next";

i18n.use(initReactI18next).init({
    resources: {
        en: enTranslations,

        ar: arTranslations,

        fr: frTranslations,
    },
    lng: "en", // Set the default language
    fallbackLng: "en",
    interpolation: {
        escapeValue: false,
    },
});

export default i18n;
