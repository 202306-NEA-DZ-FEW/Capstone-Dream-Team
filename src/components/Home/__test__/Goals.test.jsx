import { I18nextProvider } from "react-i18next";
import renderer from "react-test-renderer";

import Goals from "../Goals";
import i18n from "../../../util/i18n";

it("renders correctly", () => {
    const tree = renderer
        .create(
            <I18nextProvider i18n={i18n}>
                <Goals language='en' />
            </I18nextProvider>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
