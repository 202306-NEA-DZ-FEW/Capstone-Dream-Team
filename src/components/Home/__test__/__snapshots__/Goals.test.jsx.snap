// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="w-full h-full flex-col mt-6 p-6 justify-center  text-center"
>
  <h1
    className="font-bold mr-6 text-2xl sm:text-3xl lg:text-4xl text-[#192655] "
  >
    home.ourGoals.title
  </h1>
  <div
    className="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-3 md:gap-4 w-full h-full   "
  >
    <div
      className=" w-full h-2/3 py-4 px-8 bg-white shadow-lg rounded-lg my-20"
    >
      <div
        className="flex justify-center md:justify-end -mt-16"
      >
        <img
          alt="step"
          class="w-20 h-20 object-cover rounded-full border-2 border-indigo-500"
          src="images/home/Goals/noPoverty.png"
        />
      </div>
      <div>
        <h2
          className="text-gray-800 text-3xl font-semibold"
        >
          home.ourGoals.goal1
        </h2>
        <p
          className="mt-2 text-gray-600 h-full"
        >
          home.ourGoals.goal1text
        </p>
      </div>
    </div>
    <div
      className=" w-full h-2/3 py-4 px-8 bg-white shadow-lg rounded-lg my-20"
    >
      <div
        className="flex justify-center md:justify-end -mt-16"
      >
        <img
          alt="step"
          class="w-20 h-20 object-cover rounded-full border-2 border-indigo-500"
          src="images/home/Goals/zeroHunger.png"
        />
      </div>
      <div>
        <h2
          className="text-gray-800 text-3xl font-semibold"
        >
          home.ourGoals.goal2
        </h2>
        <p
          className="mt-2 text-gray-600 h-full"
        >
          home.ourGoals.goal2text
        </p>
      </div>
    </div>
    <div
      className=" w-full h-2/3 py-4 px-8 bg-white shadow-lg rounded-lg my-20"
    >
      <div
        className="flex justify-center md:justify-end -mt-16"
      >
        <img
          alt="step"
          class="w-20 h-20 object-cover rounded-full border-2 border-indigo-500"
          src="images/home/Goals/health.png"
        />
      </div>
      <div>
        <h2
          className="text-gray-800 text-3xl font-semibold"
        >
          home.ourGoals.goal3
        </h2>
        <p
          className="mt-2 text-gray-600 h-full"
        >
          home.ourGoals.goal3text
        </p>
      </div>
    </div>
  </div>
</div>
`;
